### ISSUE TITLE
Support custom issuers

### ISSUE THREAD

**Issue by DOMZE** [webmotions/fake-authentication-jwtbearer#1]
Right now this sets the claims with a hardcoded default [issuer](https://github.com/webmotions/fake-authentication-jwtbearer/blob/master/src/WebMotions.Fake.Authentication.JwtBearer/FakeJwtBearerClaimsHandler.cs#L67)

It would be nice if it could support custom issuers. Because sometimes in an AuthorizationHandler people sometimes do:

```csharp
var scopeClaims = context
.User
.FindAll(x => x.Type == "permissions" && x.Issuer == requirement.Issuer)
.ToList();
```

Which would fail with the fake version, since it contains "LOCAL AUTHORITY" as issuer.

**Comment by CumpsD** [comment]
This could help, it's how I fixed it in my cloned version :)

```csharp
        public virtual ClaimsIdentity CreateClaimsIdentity(Dictionary<string, JsonElement> token)
        {
            var identity = new ClaimsIdentity("FakeJwtBearer");

            foreach (var prop in token)
            {
                Claim claim;
                if (prop.Value.ValueKind == JsonValueKind.String)
                {
                    claim = CreateClaim(prop.Key, prop.Value.GetString(), identity);
                    if (claim != null)
                    {
                        identity.AddClaim(claim);
                    }
                }
                else if (prop.Value.ValueKind == JsonValueKind.Array)
                {
                    foreach (var subValue in prop.Value.EnumerateArray())
                    {
                        claim = CreateClaim(prop.Key, subValue.GetString(), identity);
                        if (claim != null)
                        {
                            identity.AddClaim(claim);
                        }
                    }
                }
                else
                {
                    throw new Exception("Unknown type");
                }
            }

            return CorrectIssuer(identity);
        }

        private static ClaimsIdentity CorrectIssuer(ClaimsIdentity identity)
        {
            var issuer = identity.FindFirst("iss");

            if (issuer == null)
                return identity;

            var newIdentity = new ClaimsIdentity("FakeJwtBearer");
            newIdentity.AddClaims(identity.Claims.Select(x => new Claim(x.Type, x.Value, x.ValueType, issuer.Value, issuer.Value, newIdentity)));
            return newIdentity;
        }
```

**Comment by DOMZE** [comment]
yea that's a good point with the iss coming from the jwt claim.

I've added the possibility to set the claim issuer/originalissuer from the iss if found and not overridden by the handler options, which gives flexibility for all sort of scenarios. If the iss is not found, I will default to what was already there, through the options.

**Comment by CumpsD** [comment]
@DOMZE looks good! :+1:

**Comment by DOMZE** [comment]
@CumpsD Fixed in 3.1.0 available for download
